### PR TITLE
Add message and monthly options to donation links

### PR DESCRIPTION
**Pull request by maxwofford**
A whole year in the making...

Screenshot 2023-07-12 at 11 40 27

Here's a preview of the UI:

https://github.com/hackclub/bank/assets/5891442/eab61316-3382-4360-beb9-e57106d37fda

### CODE REVIEW

**Comment by maxwofford** [comment]
Looks like https://github.com/hackclub/bank/issues/2943 is related.

File: app/views/donations/_donation_form.html.erb [hunk 1]
```diff
@@ -19,18 +19,26 @@
 
     <div class="field mb2">
       <%= form.label :name, "Your name" %>
-      <%= form.text_field :name, placeholder: "Fiona Hackworth", required: true, autofocus: true %>
+      <%= form.text_field :name, placeholder: "Fiona Hackworth", required: true, autofocus: true, value: params[:name] %>
```
Comment (cjdenio) [inline app/views/donations/_donation_form.html.erb]: The `value` should be set in `Donation.new` in the controller (just like `amount` and `message`)

File: app/views/donations/_donation_form.html.erb [hunk 2]
```diff
@@ -19,18 +19,26 @@
...
     </div>
 
     <div class="field mb2">
       <%= form.label :email, "Your email" %>
       <span class="muted block h5">We’ll send <%= @monthly ? "receipts" : "a receipt" %> here for your tax records.</span>
-      <%= form.email_field :email, placeholder: "fionah@gmail.com", required: true %>
+      <%= form.email_field :email, placeholder: "fionah@gmail.com", required: true, value: params[:email] %>
```
Comment (cjdenio) [inline app/views/donations/_donation_form.html.erb]: here too!

File: app/views/donations/_donation_form.html.erb [hunk 3]
```diff
@@ -19,18 +19,26 @@
...
       <% donation.errors.messages_for(:email).each do |message| %>
         <div class="primary"><%= message %></div>
       <% end %>
     </div>
 
+    <% if params[:message] %>
+      <div class="field mb2">
+        <%= form.label :message, "Message" %>
+        <span class="muted block h5">Optionally include a note.</span>
+        <%= form.text_field :message, placeholder: "Keep up the good work!", required: false, value: params[:message] %>
```
Comment (cjdenio) [inline app/views/donations/_donation_form.html.erb]: here also

**Pull request merged**